tools/SampleAnalyzer/Commons/DataFormat/WeightCollection.h

```diff
@@ -39,137 +39,205 @@
+                return false;
+            }
+
+            return true;
         }
-      }
-      catch (const std::exception &e)
-      {
-        MANAGE_EXCEPTION(e);
-        return false;
-      }
-
-      return true;
-    }
-
-    /// Get all the Weight Collection
-    const std::map<MAuint32, MAfloat64> &GetWeights() const
-    {
-      return weights_;
-    }
 
-    /// Get a weight
-    const MAfloat64 &Get(MAuint32 id) const
-    {
-      // Try to get the item
-      std::map<MAuint32, MAfloat64>::const_iterator it = weights_.find(id);
-      try
-      {
-        if (it == weights_.end())
+        /// Get all the Weight Collection
+        const std::map<MAuint32, MAfloat64> &GetWeights() const { return weights_; }
+
+        /// Get a weight
+        const MAfloat64 &Get(MAuint32 id) const
         {
-          std::stringstream str;
-          str << id;
-          std::string idname;
-          str >> idname;
-          throw EXCEPTION_ERROR("The Weight '" + idname +
-                                    "' is not defined. Return null value.",
-                                "", 0);
+            // Try to get the item
+            std::map<MAuint32, MAfloat64>::const_iterator it = weights_.find(id);
+            try
+            {
+                if (it == weights_.end())
+                {
+                    std::stringstream str;
+                    str << id;
+                    std::string idname;
+                    str >> idname;
+                    throw EXCEPTION_ERROR("The Weight '" + idname +
+                                              "' is not defined. Return null value.",
+                                          "", 0);
+                }
+            }
+            catch (const std::exception &e)
+            {
+                MANAGE_EXCEPTION(e);
+                return emptyvalue_;
+            }
+
+            return it->second;
         }
-      }
-      catch (const std::exception &e)
-      {
-        MANAGE_EXCEPTION(e);
-        return emptyvalue_;
-      }
-
-      return it->second;
-    }
-
-    /// Get a weight
-    const MAfloat64 &operator[](MAuint32 id) const
-    {
-      return Get(id);
-    }
 
-    /// Add a new weight group
-    void Print() const
-    {
-      if (weights_.empty())
-        return;
-
-      // Loop over weights for getting max
-      MAuint32 maxi = 0;
-      for (std::map<MAuint32, MAfloat64>::const_iterator
-               it = weights_.begin();
-           it != weights_.end(); it++)
-      {
-        if (it->first > maxi)
-          maxi = it->first;
-      }
-
-      // Loop over weights
-      for (std::map<MAuint32, MAfloat64>::const_iterator
-               it = weights_.begin();
-           it != weights_.end(); it++)
-      {
-        INFO << "ID=" << it->first << " : " << it->second << endmsg;
-      }
-    }
-  };
+        /// Get a weight
+        const MAfloat64 &operator[](MAuint32 id) const { return Get(id); }
+
+        /// Add a new weight group
```

Comment:
The comment on l152 is incorrect. This is the weight printing function.

Reply:
```suggestion
        /// @brief Print weight information
```